### PR TITLE
Add basic pixi support and add basic smoke test of `wholebodydynamics` device

### PR DESCRIPTION
**Pull request by traversaro**
I wanted to work on https://github.com/robotology/whole-body-estimators/issues/170 and https://github.com/robotology/whole-body-estimators/issues/176, but it was tricky to do so without any even basic test of the functionality of `wholebodydynamics` device. 

To solve the problem, I added basic support for building the project with `pixi`, and then I added a basic test that just loads the `wholebodydynamics` device file for ergoCubSN001 with fake devices.

The setup at the moment is a bit complex, as I need to compile YARP (and hence icub-main and ergocub-software) as otherwise the `fakeIMU` device in YARP 3.9 does not support specifying the `frameName`. Furthermore, I had to manually add a `fakeFTs` device to simulate a device that exposes multiple FTs sensors with temperature sensors. Once YARP 3.10 is released, the setup can be simplified to use YARP (and ICUB and ergocub-software) binaries.

### CODE REVIEW

**Comment by traversaro** [comment]
As currently building the test requires yarp master, I disable the test in CI so we can continue to use the yarp binaries there.

**Comment by traversaro** [comment]
> As currently building the test requires yarp master, I disable the test in CI so we can continue to use the yarp binaries there.

Anyhow, I added a separate pixi-ci, so conda-ci checks compilation against YARP 3.9, and pixi-ci uses instead YARP master. We can revisit this in the future.

File: README.md [hunk 1]
```diff
@@ -27,8 +29,37 @@ The current implementations available in the `devices` folder include,
...
+You can easily install `whole-body-estimators` with with `conda` using the following command
+~~~
+conda install -c conda-forge -c robotology whole-body-estimators
+~~~
+
+If you are not familiar with conda or conda-forge, you can read an introduction document in [conda-forge overview](https://github.com/robotology/robotology-superbuild/blob/master/doc/conda-forge.md#conda-forge-overview).
+
+## robotology-superbuild (advanced)
+
+If you are installing iDynTree for use as part of [iCub humanoid robot software installation](https://icub-tech-iit.github.io/documentation/sw_installation/), you may want to install iDynTree through the [robotology-superbuild](https://github.com/robotology/robotology-superbuild), an easy way to download, compile and install the robotology software on multiple operating systems, using the [CMake](https://www.cmake.org) build system and its extension [YCM](http://robotology.github.io/ycm). To get `whole-body-estimators` when using the `robotology-superbuild`, please enable the `ROBOTOLOGY_ENABLE_DYNAMICS` CMake option of the superbuild.
```
Comment (S-Dafarra) [inline README.md:45]:
```suggestion
If you are installing `whole-body-estimators` for use as part of [iCub humanoid robot software installation](https://icub-tech-iit.github.io/documentation/sw_installation/), you may want to install `whole-body-estimators` through the [robotology-superbuild](https://github.com/robotology/robotology-superbuild), an easy way to download, compile and install the robotology software on multiple operating systems, using the [CMake](https://www.cmake.org) build system and its extension [YCM](http://robotology.github.io/ycm). To get `whole-body-estimators` when using the `robotology-superbuild`, please enable the `ROBOTOLOGY_ENABLE_DYNAMICS` CMake option of the superbuild. 
```

File: cmake/WBEAddUnitTest.cmake [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+    set(targetname ${name}UnitTests)
+    add_executable(${targetname}
+        "${unit_test_files}")
+
+    target_link_libraries(${targetname} PRIVATE Catch2::Catch2WithMain ${${prefix}_LINKS})
+    target_compile_definitions(${targetname} PRIVATE CATCH_CONFIG_FAST_COMPILE CATCH_CONFIG_DISABLE_MATCHERS)
+
+    add_test(NAME ${targetname} COMMAND ${targetname})
+endfunction()
```
Comment (S-Dafarra) [inline cmake/WBEAddUnitTest.cmake:25]:
```suggestion
endfunction()

```

File: devices/wholeBodyDynamics/test/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+add_subdirectory(fakeFTs)
+
+wbe_add_unit_test(NAME WholeBodyDynamics
+                  SOURCES WholeBodyDynamicsUnitTests.cpp
+                  LINKS YARP::YARP_robotinterface)
+
+target_compile_definitions(WholeBodyDynamicsUnitTests 
+                           PRIVATE CMAKE_CURRENT_SOURCE_DIR="${CMAKE_CURRENT_SOURCE_DIR}"
+                                   PROJECT_BINARY_DIR="${PROJECT_BINARY_DIR}")
```
Comment (S-Dafarra) [inline devices/wholeBodyDynamics/test/CMakeLists.txt:11]:
```suggestion
                                   PROJECT_BINARY_DIR="${PROJECT_BINARY_DIR}")

```

File: set_install_prefix.sh [hunk 4]
```diff
@@ -0,0 +1,2 @@
+export CMAKE_INSTALL_PREFIX=$CONDA_PREFIX
+export YARP_DATA_DIRS=$CMAKE_INSTALL_PREFIX/share/yarp:$CMAKE_INSTALL_PREFIX/share/ergoCub
```
Comment (S-Dafarra) [inline set_install_prefix.sh:2]:
```suggestion
export YARP_DATA_DIRS=$CMAKE_INSTALL_PREFIX/share/yarp:$CMAKE_INSTALL_PREFIX/share/ergoCub

```

File: devices/wholeBodyDynamics/test/WholeBodyDynamicsUnitTests.cpp [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    // Enter the startup phase, that will open all the devices and  call attach if necessary
+    REQUIRE(xmlRobotInterfaceResult.robot.enterPhase(yarp::robotinterface::ActionPhaseStartup));
+    REQUIRE(xmlRobotInterfaceResult.robot.enterPhase(yarp::robotinterface::ActionPhaseRun));
+
+    // If the robotinterface started, we are good to go, we can close it 
+    REQUIRE(xmlRobotInterfaceResult.robot.enterPhase(yarp::robotinterface::ActionPhaseInterrupt1));
+    REQUIRE(xmlRobotInterfaceResult.robot.enterPhase(yarp::robotinterface::ActionPhaseInterrupt2));
+    REQUIRE(xmlRobotInterfaceResult.robot.enterPhase(yarp::robotinterface::ActionPhaseInterrupt3));
+    REQUIRE(xmlRobotInterfaceResult.robot.enterPhase(yarp::robotinterface::ActionPhaseShutdown));
+}
```
Comment (S-Dafarra) [inline devices/wholeBodyDynamics/test/WholeBodyDynamicsUnitTests.cpp:62]:
```suggestion
}

```

File: devices/wholeBodyDynamics/test/ergocub_test_all.xml [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    <!-- Simulated IMUs -->
+    <device name="simulated_imu" type="fakeIMU">
+        <param name="sensorName">"waist_imu_0"</param>
+        <param name="frameName">"waist_imu_0"</param>
+    </device>
+
+    <!-- Simulated WBD -->
+    <xi:include href="./ergocub_simulated_wholebodydynamics.xml" />
+  </devices>
+</robot>
```
Comment (S-Dafarra) [inline devices/wholeBodyDynamics/test/ergocub_test_all.xml:28]:
```suggestion
</robot>

```

File: devices/wholeBodyDynamics/test/fakeFTs/CMakeLists.txt [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+    YARP::YARP_dev
+    YARP::YARP_math
+)
+
+yarp_install(TARGETS fakeFTs
+             EXPORT Runtime
+             COMPONENT runtime
+             LIBRARY DESTINATION ${YARP_DYNAMIC_PLUGINS_INSTALL_DIR}
+             ARCHIVE DESTINATION ${YARP_STATIC_PLUGINS_INSTALL_DIR}
+             YARP_INI DESTINATION ${YARP_PLUGIN_MANIFESTS_INSTALL_DIR})
```
Comment (S-Dafarra) [inline devices/wholeBodyDynamics/test/fakeFTs/CMakeLists.txt:29]:
```suggestion
             YARP_INI DESTINATION ${YARP_PLUGIN_MANIFESTS_INSTALL_DIR})

```

**Review by S-Dafarra — APPROVED** [review]
Nice! Only minor comments

File: README.md (hunk 1 above)
Comment (traversaro) [reply README.md:45]: I think you found where I copied and pasted the docs. :D

**Pull request merged**